clippy_lints/src/methods/map_clone.rs

```diff
@@ -86,8 +86,11 @@ pub(super) fn check(cx: &LateContext<'_>, e: &hir::Expr<'_>, recv: &hir::Expr<'_
                                     }
                                 }
                             },
-                            hir::ExprKind::Call(call, [_]) => {
-                                if let hir::ExprKind::Path(qpath) = call.kind {
+                            hir::ExprKind::Call(call, args) => {
+                                if let hir::ExprKind::Path(qpath) = call.kind
+                                    && let [arg] = args
+                                    && ident_eq(name, arg)
+                                {
```

Comment:
By the way, if type checking succeeds and `name` == `arg` (e.g., 
```
let x: Option<&u8> = None;
let y = x.map(|x| u8::clone(x));
//             ^ name       ^ arg (in my understanding)
```
), it seems like the modification I did in `handle_path(...)` is not necessary.

What do you think?

Reply:
I've remove the checks and it looks everything passes.